winterwell.datalog/src/java/com/winterwell/datalog/server/CompressDataLogIndexMain.java

```diff
@@ -0,0 +1,202 @@
+	
+	public final static String ALIAS = "datalog.transformed.all";
+	
+	
+	public static void main(String[] args) {
+		CompressDataLogIndexMain amain = new CompressDataLogIndexMain();
+		amain.doMain(args);		
+	}
+
+	@Override
+	protected void init2(DataLogConfig config) {
+		logFile = new LogFile(FileUtils.changeType(config.logFile, "compressor.txt"))
+				// keep 1 week of log files
+				.setLogRotation(TUnit.DAY.dt, 7);
+		// set the config
+		DataLog.init(config);
+		// usual setup
+		super.init2(config);
+		
+		init3_ES();
+//		init3_gson();
+//		init3_youAgain();		
+	}
+
+	
+	/**
+	 * Maybe get from config.namespace??
+	 */
+	private String dataspace = "gl";
+	
+	public CompressDataLogIndexMain() {
+		super("CompressDataLogIndex", DataLogConfig.class);
+	}
+	
+	@Override
+	protected void doMain2() {
+		// which month? (3 letter lowercase) and year (last 2 digits)		
+		Time t;
+		if ( ! Utils.isEmpty(configRemainderArgs)) {
+			t = TimeUtils.parseExperimental(configRemainderArgs.get(0));
+		} else {
+			t = new Time().minus(2, TUnit.MONTH);
+		}		
+		String monthYear = t.format("MMMyy").toLowerCase();
+		String index = "datalog."+dataspace+"_transformed_" + monthYear;
+		String source = "datalog."+dataspace+"_" + monthYear;
+
+		// specify some terms that we want to keep
+		// See DataLogEvent#COMMON_PROPS
+		// TODO increase this list as our usage changes
+		
+		List<String> aggs = Arrays.asList(("amount dntn").split(" "));
+		List<String> terms = Arrays.asList(
+				("evt domain host country pub vert vertiser campaign lineitem "
+				 +"cid via invalid mbl browser os"
+				).split(" ")
+		);
+
+		// create index and mapping
+		createIndexWithPropertiesMapping(index, ALIAS, terms);
+		
+		ESHttpClient esc = Dep.get(ESHttpClient.class);
+		// aggregate data
+		String jobId = "transform_"+dataspace+"_"+monthYear;
+		
+		//safety mechanism -- make sure that the jobID doens't exist, if it does, delete it
+		try {
+			TransformRequestBuilder trb_safety = esc.prepareTransformDelete(jobId); 
+			trb_safety.setDebug(true);
+			trb_safety.get();
+		} catch (ESDocNotFoundException e) {
+			Printer.out("Safe to continue...");
+		}
+		
+		// create transform job
+		// specify source and destination and time interval
+		TransformRequestBuilder trb = esc.prepareTransform(jobId);
+		trb.setBody(source, index, aggs, terms, "24h");
+		//trb.setBodyWithPainless(source, index, aggs, terms, "24h"); //for ES version < 7.10.0
+		trb.setDebug(true);
+		IESResponse response = trb.get(); //might take a long time for complex body
+		Log.d("compress", response);
+		
+		//after creating transform job, start it 
+		TransformRequestBuilder trb2 = esc.prepareTransformStart(jobId); 
+		trb2.setDebug(true);
+		IESResponse response2 = trb2.get().check();
+		Log.d("compress", response2);
+		System.out.println("Transforming data, please wait...");
+		
+		Utils.sleep(36000000); //allow transform job to be completed before deleting it, a safe estimate would be one hour
```

Comment:
Hm... A small index should be fast, and a shame to take an hour. And maybe a monster index and/or an overloaded server could take longer than an hour.

I bet there's an API method to check the status.

Let's poll for status every 5 seconds if we can.